engine/src/main/java/org/hibernate/search/engine/backend/document/converter/ToIndexIdValueConverter.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.backend.document.converter;
+
+import org.hibernate.search.engine.backend.document.converter.runtime.ToIndexFieldValueConvertContext;
+import org.hibernate.search.engine.backend.document.converter.runtime.ToIndexIdValueConvertContext;
+import org.hibernate.search.engine.backend.document.converter.runtime.ToIndexIdValueConvertContextExtension;
+
+/**
+ * A converter from a source value to a target value that should be indexed.
+ *
+ * @param <V> The type of source values.
+ */
+public interface ToIndexIdValueConverter<I> {
+
+	/**
+	 * @param value The source value to convert.
+	 * @param context A context that can be
+	 * {@link ToIndexIdValueConvertContext#extension(ToIndexIdValueConvertContextExtension) extended}
+	 * to a more useful type, giving access to such things as a Hibernate ORM SessionFactory (if using the Hibernate ORM mapper).
+	 * @return The converted index field value.
+	 */
+	default String convert(I value, ToIndexIdValueConvertContext context) {
+		if ( value == null ) {
+			return null;
+		}
+		return String.valueOf( value );
```

Comment:
Not sure I would provide a default implementation here, especially since this uses `toString`, which generally should be considered as a method returning a human-readable representation of the object, and thus is not appropriate in this context.